modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -338,6 +346,27 @@ object ConsensusStateUpdater {
     private def pickMajority[A: Order](proposals: List[A]): Option[A] =
       proposals.foldMap(a => Map(a -> 1)).toList.map(_.swap).maximumOption.map(_._2)
 
+    private def pickValidatedMajority(
+      lastSignedArtifact: Signed[Artifact],
+      trigger: ConsensusTrigger,
+      resources: ConsensusResources[Artifact]
+    )(proposals: List[Hash]): F[Option[Hash]] =
+      proposals
+        .foldMap(a => Map(a -> 1))
+        .toList
+        .map(_.swap)
+        .sortBy(_._1)
```

Comment:
1. This sorts only by `count` (occurrences). While it should sort by both `(count, hash)`
2. Default sort order is ascending, so aren't you taking the value with the smallest count first?

Reply:
good catch :+1: